tests/ui/stacked_ifs.rs

```diff
@@ -0,0 +1,53 @@
+#![warn(clippy::stacked_ifs)]
+
+fn main() {
+    let a = 3;
+    let b = 4;
+    if a == b { // No error
+        // Do something.
+    } else {
+        // Do something else.
+    }
+
+    if if a % 2 == 0 {
+        //~^ ERROR: stacked `if` found
+        a - 1
+    } else {
+        a + 1
+    } == b
+    {
+        // Do something.
+    }
+
+    if if a % 2 == 0 {
+        //~^ ERROR: stacked `if` found
+        a - 1
+    } else {
+        a + 1
+    } == b
+    {
+        // Do something.
+    } else {
+        // Do something else.
+    }
+
+    let c = 5;
+    let d = 6;
+
+    // More complex statements (example from issue 12483):
+    if if if a == b {
+        //~^ ERROR: stacked `if` found
+        //~^^ ERROR: stacked `if` found
+        b == c
+    } else {
+        a == c
+    } {
+        a == d
+    } else {
+        c == d
+    } {
+        println!("True!");
+    } else {
+        println!("False!");
+    }
+}
```

Comment:
A test for `if if let ...` would be good

Reply:
As would a base case without a binary expression

```rust
if if true { true } else { false } {}
```

If other lints get triggered by this add them to an `#![allow]`